tests/event/test_event_reorg.py

```diff
@@ -0,0 +1,59 @@
+from hathor.conf import HathorSettings
+from tests import unittest
+from tests.utils import add_new_blocks, get_genesis_key
+
+settings = HathorSettings()
+
+
+class BaseEventReorgTest(unittest.TestCase):
+    __test__ = False
+
+    def setUp(self):
+        super().setUp()
+        self.network = 'testnet'
+        self.manager = self.create_peer(self.network, event_storage=True)
+        self.event_manager = self.manager.event_manager
+        self.event_storage = self.event_manager.event_storage
+
+        # read genesis keys
+        self.genesis_private_key = get_genesis_key()
+        self.genesis_public_key = self.genesis_private_key.public_key()
+
+    def test_reorg_events(self):
+        # add some blocks
+        blocks = add_new_blocks(self.manager, settings.REWARD_SPEND_MIN_BLOCKS, advance_clock=1)
+
+        # make a re-org
+        self.log.debug('make reorg block')
+        block_to_replace = blocks[-2]
+        tb0 = self.manager.make_custom_block_template(block_to_replace.parents[0], block_to_replace.parents[1:])
+        b0 = tb0.generate_mining_block(self.manager.rng, storage=self.manager.tx_storage)
+        b0.weight = 10
+        b0.resolve()
+        b0.verify()
+        self.manager.propagate_tx(b0, fails_silently=False)
+        self.log.debug('reorg block propagated')
+        self.run_to_completion()
+
+        # check events
+        event_count = self.event_storage.get_last_event().id + 1
+        events = []
+        for i in range(event_count):
+            events.append(self.event_storage.get_event(i))
+
+        for ev in events:
+            print(ev.type)
+            # TODO
```

Comment:
Should we assert that the correct events were generated? Maybe compare it with a list of expected event types?

Reply:
Yes, exactly, I missed this TODO.

Reply:
Done.

Reply:
Great! Thanks!